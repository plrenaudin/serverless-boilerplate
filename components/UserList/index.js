import PropTypes from "prop-types";
import React from "react";
import fetcher from "../../modules/fetcher";
import "./index.less";

const deleteUser = id => () => {
  fetcher(`/api/users/${id}`, {
    method: "DELETE"
  }).then(() => {
    //Hack, should be handled by a global state management or something...
    window.location.reload();
  });
};

const UserList = ({ users }) => (
  <>
    {users.length > 0 && (
      <ul>
        {users.map(i => (
          <li key={i._id}>
            {i.name} {i.age}
            <button onClick={deleteUser(i._id)}>Delete</button>
          </li>
        ))}
      </ul>
    )}
  </>
);

UserList.propTypes = {
  users: PropTypes.array.isRequired
};

export default React.memo(UserList);
