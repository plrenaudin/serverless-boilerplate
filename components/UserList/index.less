.userList {
  ul {
    list-style-type: none;
    width: 500px;
    margin: 0;
    padding: 0;
    li {
      display: flex;
      padding: 5px;
      justify-content: space-between;
      &:nth-child(odd) {
        background: #f3f3f3;
      }
      &:hover {
        background-color: #bbdefb;
        cursor: pointer;
      }
    }
  }
}
