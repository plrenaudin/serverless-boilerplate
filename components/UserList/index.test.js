import renderer from "react-test-renderer";
import UserList from ".";

it("renders correctly", () => {
  const users = [
    { _id: 1, name: "Mickey", age: 15 },
    { _id: 2, name: "Dan", age: 25 },
    { _id: 3, name: "John", age: 42 }
  ];
  const tree = renderer.create(<UserList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly without data", () => {
  const users = [];
  const tree = renderer.create(<UserList users={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});
