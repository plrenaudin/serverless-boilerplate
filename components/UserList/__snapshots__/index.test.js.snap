// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="userList"
>
  <ul>
    <li>
      Mickey
       
      15
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </li>
    <li>
      Dan
       
      25
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </li>
    <li>
      John
       
      42
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </li>
  </ul>
</section>
`;

exports[`renders correctly without data 1`] = `
<section
  className="userList"
/>
`;
